db/db_iter.cc

```diff
@@ -933,12 +933,7 @@ void DBIter::FindPrevUserKey() {
     return;
   }
   size_t num_skipped = 0;
-// Suppress false positive clang analyzer warnings.
-#ifdef __clang_analyzer__
-  ParsedInternalKey ikey(Slice(), 0, 0);
```

Comment:
My bad. In #2334 I add these because clang-analyzer complain about `ikey` is not initialized properly, but leave the cast error there. It didn't trigger any build error on my devserver. Can you change this line to `ParsedInternalKey ikey(Slice(), 0, kMaxValue);` instead?